aws/internal/service/elasticache/finder/finder.go

```diff
@@ -23,23 +25,21 @@ func ReplicationGroupByID(conn *elasticache.ElastiCache, id string) (*elasticach
 		return nil, err
 	}
 
-	if result == nil || len(result.ReplicationGroups) == 0 || result.ReplicationGroups[0] == nil {
+	if output == nil || len(output.ReplicationGroups) == 0 || output.ReplicationGroups[0] == nil {
 		return nil, &resource.NotFoundError{
 			Message:     "Empty result",
 			LastRequest: input,
 		}
 	}
 
-	return result.ReplicationGroups[0], nil
+	return output.ReplicationGroups[0], nil
 }
 
 // ReplicationGroupMemberClustersByID retrieves all of an ElastiCache Replication Group's MemberClusters by the id of the Replication Group.
 func ReplicationGroupMemberClustersByID(conn *elasticache.ElastiCache, id string) ([]*elasticache.CacheCluster, error) {
-	var results []*elasticache.CacheCluster
-
 	rg, err := ReplicationGroupByID(conn, id)
 	if err != nil {
-		return results, err
+		return []*elasticache.CacheCluster{}, err
```

Comment:
Nit: Could return `nil` here instead of a 0 element slice.

```suggestion
		return nil, err
```